service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -97,13 +100,25 @@ public UUID createWorkspace(Workspace workspace) {
       jdbcTemplate.update(sql, params);
       logger.info("Inserted record for workspace {}", workspaceId);
     } catch (DuplicateKeyException e) {
-      throw new DuplicateWorkspaceException(
-          String.format(
-              "Workspace with id %s already exists - display name %s stage %s",
-              workspaceId,
-              workspace.getDisplayName().toString(),
-              workspace.getWorkspaceStage().toString()),
-          e);
+      // Workspace with workspace_id already exists.
+      if (e.getMessage()
+          .contains("duplicate key value violates unique constraint \"workspace_pkey\"")) {
+        throw new DuplicateWorkspaceException(
+            String.format(
+                "Workspace with id %s already exists - display name %s stage %s",
+                workspaceId,
+                workspace.getDisplayName().toString(),
+                workspace.getWorkspaceStage().toString()),
+            e);
+      } else {
```

Comment:
suggestion: don't assume its the ufid key that fails here as 'else', and have the 'else' be a fallback.